homeassistant/components/lg_thinq/sensor.py

```diff
@@ -0,0 +1,590 @@
+"""Support for sensor entities."""
+
+from __future__ import annotations
+
+import logging
+
+from thinqconnect import DeviceType
+from thinqconnect.devices.const import Property as ThinQProperty
+from thinqconnect.integration import ThinQPropertyEx, TimerProperty
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+    PERCENTAGE,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ThinqConfigEntry
+from .coordinator import DeviceDataUpdateCoordinator
+from .entity import ThinQEntity
+
+COMMON_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQPropertyEx.ERROR: SensorEntityDescription(
+        key=ThinQPropertyEx.ERROR,
+        translation_key=ThinQPropertyEx.ERROR,
+    ),
+    ThinQPropertyEx.NOTIFICATION: SensorEntityDescription(
+        key=ThinQPropertyEx.NOTIFICATION,
+        translation_key=ThinQPropertyEx.NOTIFICATION,
+    ),
+}
+AIR_QUALITY_SENSOR_DESC: dict[ThinQProperty, SensorEntityDescription] = {
+    ThinQProperty.PM1: SensorEntityDescription(
+        key=ThinQProperty.PM1,
+        device_class=SensorDeviceClass.PM1,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM1,
+    ),
+    ThinQProperty.PM2: SensorEntityDescription(
+        key=ThinQProperty.PM2,
+        device_class=SensorDeviceClass.PM25,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM2,
+    ),
+    ThinQProperty.PM10: SensorEntityDescription(
+        key=ThinQProperty.PM10,
+        device_class=SensorDeviceClass.PM10,
+        native_unit_of_measurement=CONCENTRATION_MICROGRAMS_PER_CUBIC_METER,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.PM10,
+    ),
+    ThinQProperty.HUMIDITY: SensorEntityDescription(
+        key=ThinQProperty.HUMIDITY,
+        device_class=SensorDeviceClass.HUMIDITY,
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+        translation_key=ThinQProperty.HUMIDITY,
+    ),
```

Comment:
All these entities with a device class which is not enum (where the name is the same as the device class), can be removed from strings.json and icons.json, as they get their name from the device class

Reply:
Updated